lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -30,7 +32,11 @@ def execute(self):
         except requests.exceptions.ConnectionError as exc:
             raise RuntimeError(f"Cannot connect to '{uri}'") from exc
 
-        payload = response.json()
+        try:
+            payload = response.json()
+        except JSONDecodeError:
+            payload = {"message": response.text}
+
         if response.ok:
             click.echo(payload["auth_token"])
```

Comment:
This is interesting, because if we get the `JSONDecodeError` when `response.ok`, we'll get a key error accessing `"auth_token"`, which probably isn't ideal.

However, for my case, with the server stopped, this is a distinct improvement, although the rendering of `response.text` (which is apparently an Apache error html page) is pretty ugly:

```
[root@dhcp31-136 ~]# pbench-generate-token --username roger --password roger
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>503 Service Unavailable</title>
</head><body>
<h1>Service Unavailable</h1>
<p>The server is temporarily unable to service your
request due to maintenance downtime or capacity
problems. Please try again later.</p>
</body></html>
```

I wonder if a `try` around the `payload["auth_token"]` to cleanly handle an error there might not be a bad idea while you're at it?

Reply:
Instead of `response.text` I could use `response.reason` (which, in the case, above, I presume would be `Service Unavailable`), do you think that would be better?

(I'll see what I can do for the `KeyError` problem.)

Reply:
OK, in the unlikely event that the response is "OK" but doesn't return an `"auth_token"`, we will now exit with an error after reporting that things are "OK".  😝

Let me know if this is not Good-er.